packages/workbox-background-sync/src/lib/request-queue.js

```diff
@@ -96,6 +96,7 @@ class RequestQueue {
    * @private
    */
   async push({request}) {
+    await this._initializationPromise;
```

Comment:
@prateekbh Do what ever you think it best. It was more of a suggestion based off of reading the current code which references the queue multiple times and that we already missed a place where the initialisation promise should have been waited on but wasn't because it's not obvious where it is needed. A getQueue() method for all uses of the queue will force the waiting on the initialisation without any real thought from the developers of this module.